src/Paramore.Brighter.ServiceActivator/MessagePump.cs

```diff
@@ -132,17 +151,21 @@ public void Run()
                         Channel.Name, Thread.CurrentThread.ManagedThreadId);
 
                     RejectMessage(message);
+                    span?.SetStatus(ActivityStatusCode.Error,
```

Comment:
I wonder if this would be part of a span? Is the health of this serviceactivator part of a trace?

Reply:
As with the other comment. let's see what this looks like. I guess it is conceptual, is a trace intended to 'monitor' health, or just to show 'flow' and events in the flow. Is a 'stop' event, just in the flow, perhaps, but not sure that 'shutdown' subsequent is for that trace.

Reply:
As discussed, this just shows as a failure of processing of that one part of the trace